azure/Guidelines.md

```diff
@@ -1279,6 +1279,31 @@ In addition to distributed tracing, Azure also uses a set of common correlation
 |x-ms-client-request-id       |Both      |Optional. Caller-specified value identifying the request, in the form of a GUID with no decoration such as curly braces (e.g. `x-ms-client-request-id: 9C4D50EE-2D56-4CD3-8152-34347DC9F2B0`). If the caller provides this header the service **must** include this in their log entries to facilitate correlation of log entries for a single request. Because this header can be client-generated, it should not be assumed to be unique by the service implementation.
 |x-ms-request-id              |Response  |Required. Service generated correlation id identifying the request, in the form of a GUID with no decoration such as curly braces. In contrast to the the `x-ms-client-request-id`, the service **must** ensure that this value is globally unique. Services should log this value with their traces to facilitate correlation of log entries for a single request.
 
+<a name="serversentevents"></a>
+### Server Sent Events (SSE)
+For services that generate partial results over time, it is often beneficial to return the data as soon as it is available to reduce latency. A specific example of this is the generate method for Open AI. One option for this is [Server Sent Events (SSE)](https://html.spec.whatwg.org/multipage/server-sent-events.html#server-sent-events). SSE may be used under the following conditions:
```

Comment:
* Do we want to mention some limitations of SSE for the service designer to be aware of? For example, the connection limit per browser, or some proxies do not support SSE. 
* Usually there might be some keep-alive mechanism for the SSE connection to detect if the underlying connection is dropped. For example when the network cable is unplugged, the TCP connection is still considered as connected for minutes.
*  Is it better for the service to always provide a fallback plan (e.g. longpolling) so that the client can switch to it when SSE does not work?

Reply:
I assume that implementors are already familiar with SSE and its inherent limitations, so I wasn't intending to repeat that here. If we want to expand on the guidance, we can do so. 

Regarding keep-alive, the intent of limiting usage to relatively short sessions is to minimize the need. The standard keep-alive (e.g. occasionally sending comments) is something that I also assumed that service implementors are aware of. But, again, we can add this.

I would not recommend always having a fallback plan like long polling. Instead I'd recommend that there is a non-streaming way to get the same data once it is all available.